File: Task2/Models/Study.swift
//
//  Study.swift
//  Task2
//
//  Created by skia mac mini on 5/8/24.
//

import Foundation

struct Study: Decodable {
    let id: Int
    let studyInstanceUID: String
    let studyID: String
    let studyDateTime: Date
    let studyDescription: String
    let patientID: String
    let patientName: String
    let patientBirthDate: Date
    let patientSex: String
    let referringPhysicianName: String
    let numberOfSeries: Int
    let createdDateTime: Date
    let lastModifiedDateTime: Date
    let tag: String
    let memo: String
}

File: server/DicomServer/Models/Study.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DicomServer.Models
{
    public struct Study
    {
        int id { get; }
        string studyInstanceUID { get; }
        string studyID { get; }
        string studyDateTime { get; }
        string studyDescription { get; }
        string patientID { get; }
        string patientName { get; }
        string patientBirthDate { get; }
        string patientSex { get; }
        string ? referringPhysicianName { get; }
        int numberOfSeries { get; }
        string createdDateTime { get; }
        string lastModifiedDateTime { get; }
        string ? tag { get; }
        string ? memo { get; }

    }
}